react/training/src/redux/Auth/action.js

```diff
@@ -0,0 +1,42 @@
+import AuthService from '~services/AuthService';
+
+const privateActions = {
+  requestHasError: isError => ({
+    type: 'REQUEST_HAS_ERRORED',
+    hasError: isError
+  }),
+  assignLoading: loading => ({
+    type: 'ASSIGN_LOADING',
+    loading
+  })
+};
+
+const actionCreators = {
+  authUser: currentUser => dispatch => {
+    dispatch(privateActions.assignLoading(true));
+    AuthService.getUsers()
+      .then(response => {
+        if (!response.ok) {
+          throw Error(response.statusText);
+        }
+        return response.data;
+      })
+      .then(data => {
+        dispatch(privateActions.assignLoading(false));
+        const userExist = data.find(
+          user => user.email === currentUser.email && user.password === currentUser.password
+        );
+        if (userExist) {
+          localStorage.setItem('token', userExist.token);
+          AuthService.setToken(userExist.token);
+          dispatch(privateActions.requestHasError(false));
+          dispatch({ type: 'AUTH_USER', auth: true });
```

Comment:
`AUTH_USER` must be a private actions and the payload of this actions must be the user data.